sklearn/experimental/tests/test_enable_hist_gradient_boosting.py

```diff
@@ -0,0 +1,103 @@
+import tempfile
+from subprocess import check_output, STDOUT, CalledProcessError
+import os
+import os.path as op
+import sys
+import textwrap
+from subprocess import TimeoutExpired
+
+
+TIMEOUT = 60
+
+
+def _make_cwd_env():
+    """Helper to prepare environment for the child processes"""
+    # This is a port from cloudpickle https://github.com/cloudpipe/cloudpickle
+    sklearn_repo_folder = op.normpath(
+        op.join(op.dirname(__file__), '../..'))
+    env = os.environ.copy()
+    pythonpath = "{src}{sep}tests{pathsep}{src}".format(
+        src=sklearn_repo_folder, sep=os.sep, pathsep=os.pathsep)
+    env['PYTHONPATH'] = pythonpath
+    return sklearn_repo_folder, env
+
+
+def assert_run_python_script(source_code, timeout=TIMEOUT):
```

Comment:
Maybe this should be moved to `sklearn.utils.testing` where all the `assert_*` helpers live.

Reply:
I moved it there but I think it's making the CI go bad with the coverage. https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=2330&view=logs

Any idea why this could happen?

Reply:
The error message hints that we collect coverage for the subprocess in a different manner than for the main pytest process. coverage can be either computed in a line by line basis or a branch by branch basis. We have a `.coveragerc` file with the following content in scikit-learn:

```ini
[run]
branch = True
source = sklearn
include = */sklearn/*
omit =
    */sklearn/externals/*
    */benchmarks/*
    */setup.py
```

Maybe we could be more explicit and pass: the `--cov-file=$BUILD_SOURCESDIRECTORY/.coveragerc` flag to our pytest commands on azure and also defining the following environment variable would help make everything more consistent?

```bash
COVERAGE_PROCESS_START="$BUILD_SOURCESDIRECTORY/.coveragerc"
```

What I do not understand is why the location of the helper function has an impact.

Reply:
If we fix the CWD as mentioned in https://github.com/scikit-learn/scikit-learn/pull/12807#pullrequestreview-228880168, passing `--cov-file` and `COVERAGE_PROCESS_START` is probably not necessary although it would probably won't hurt (if configured properly).